ursh/indico_ursh/client/index.js

```diff
@@ -0,0 +1,122 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {handleAxiosError, indicoAxios} from 'indico/utils/axios';
+
+
+async function _makeUrshRequest(originalURL) {
+    const urshEndpoint = '/ursh';
+
+    let response;
+    try {
+        response = await indicoAxios.post(urshEndpoint, {
+            original_url: originalURL,
+        });
+    } catch (error) {
+        handleAxiosError(error);
+        return;
+    }
+
+    return response.data.url;
+}
+
+function _validateAndFormatURL(url) {
+    const $t = $T.domain('ursh');
+    if (!url) {
+        throw $t.gettext('Please fill in a URL to shorten');
+    }
+
+    // if protocol is missing, prepend it
+    if (url.startsWith(window.location.hostname)) {
+        url = `${window.location.protocol}//${url}`;
+    }
+
+    let parsedURL;
+    try {
+        parsedURL = new URL(url);
```

Comment:
https://caniuse.com/#feat=url :crying_cat_face:

Reply:
What is the problem? Too small coverage? IE?

Reply:
IE11.. and I guess a ponyfill for it is quite big?

Reply:
I guess I could use regular expressions, but this seemed cleaner and less error-prone...